### PR TITLE
[1.0] Redux example

### PR DESCRIPTION
**Pull request by scottyeck**
Related: https://github.com/gatsbyjs/gatsby/issues/1079

### CODE REVIEW

File: examples/redux/src/html.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    return (
+      <html op="news" lang="en">
+        <head>
+          {this.props.headComponents}
+
+          <meta name="referrer" content="origin" />
+          <meta charSet="utf-8" />
+          <meta
+            name="description"
+            content="Gatsby example site showing use with no plugins"
```
Comment (KyleAMathews) [inline examples/redux/src/html.js]: this and title need updated

**Comment by KyleAMathews** [comment]
Nice! Perfect little example. Very clear.

**Comment by mericsson** [comment]
Agree, great example! Thanks @scottyeck!

@KyleAMathews I used this to see I could reproduce the issue I was seeing where webpack was not getting applied to `gatsby-ssr.js`. I wonder if @scottyeck sees this as well?

I pulled the code out into a [separate repo](https://github.com/mericsson/gatsby-using-redux) to make it easier to repro.

`yarn run build` && `yarn run serve-build` works fine and the webapp renders correctly.

`yarn run develop` builds fine, but when rendering http://localhost:9000 I get:

```
/Users/mericsson/dev/gatsby-using-redux/gatsby-ssr.js:1
(function (exports, require, module, __filename, __dirname) { import React from 'react'
                                                              ^^^^^^
SyntaxError: Unexpected token import
    at Object.exports.runInThisContext (vm.js:76:16)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/mericsson/dev/gatsby-using-redux/.cache/api-runner-ssr.js:2:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
```

@KyleAMathews If it repro's for you too, I can look into a fix if you point me in the right direction.

**Comment by KyleAMathews** [comment]
ahhh this very specific case is where there's trouble because Gatsby isn't running things through Babel or Webpack. Gatsby does a server render of your html.js/ssr plugins when you load a page in development https://github.com/gatsbyjs/gatsby/blob/4645524d3d1942d4861999536ae2156aaa12d314/packages/gatsby/src/utils/develop.js#L103

To ensure consistency with production, we should really be generating an HTML file there with https://github.com/markdalgleish/static-site-generator-webpack-plugin Right now a gatsby-ssr plugin could import a file or some other webpack-specific thing and development would break.

Would be great if you could tackle that.

@scottyeck on this PR — do you think you'll have a bit of time in the next day or two to push this across the finish line?

**Comment by scottyeck** [comment]
@mericsson Yup - can confirm that we're seeing the same thing. We had to ship something ASAP, so I actually just converted all of our `gatsby-ssr` dependencies (there weren't many) to support `require`. Would love if I didn't have to do that extra brainwork.

**Comment by KyleAMathews** [comment]
Thanks @scottyeck !

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit b27a1fac20bc135c6209b788e4ad45f97ab28742

https://app.netlify.com/sites/using-styled-components/deploys/593875218ebdd94fe24352d3

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit b27a1fac20bc135c6209b788e4ad45f97ab28742

https://app.netlify.com/sites/using-glamor/deploys/593875218ebdd94fe24352d0

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit b27a1fac20bc135c6209b788e4ad45f97ab28742

https://app.netlify.com/sites/using-contentful/deploys/593875218ebdd94fe24352d6

**Comment by KyleAMathews** [comment]
Not quite sure what those netlify errors are but they don't have anything to do with this PR

**Pull request merged**